src/server/nbs/nbs_play_action.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef NBS_PLAY_ACTION_H
+#define NBS_PLAY_ACTION_H
+
+#include <condition_variable>
+#include <nan.h>
+#include <memory>
+#include <mutex>
+#include "nbs_player.hpp"
+
+class NBSPlayAction : public Nan::AsyncProgressQueueWorker<Packet> {
+public:
+    NBSPlayAction(NBSPlayer* player);
+
+    void Execute(const ExecutionProgress& p);
+
+    void HandleProgressCallback(const Packet*, size_t);
+
+    void HandleOKCallback();
+
+    void HandleErrorCallback();
+
+    NBSPlayer* player;
```

Comment:
Cyclic dependencies are bad mmkay.
Maybe some sort of lambda based system would work here.